src/panels/energy/ha-panel-energy.ts

```diff
@@ -118,6 +136,15 @@ class PanelEnergy extends LitElement {
     };
   }
 
+  private _toggleCompare() {
+    this._compare = !this._compare;
+    const energyCollection = getEnergyDataCollection(this.hass, {
+      key: "energy_dashboard",
+    });
+    energyCollection.setCompare(this._compare);
+    energyCollection.refresh();
```

Comment:
We only need to refresh if compare is now true and it's not loaded for current period yet?